lib/datadog/lambda/trace/ddtrace.rb

```diff
@@ -29,14 +30,15 @@ def apply_datadog_trace_context(context)
       def wrap_datadog(options, &block)
         tracer_available = false
         begin
-          Datadog.tracer
+          raise StandardError unless Datadog::Tracing.enabled?
```

Comment:
Hmmm if we're going to raise our own error, why not use a conditional instead of a heavier-weight exception?

Reply:
Updated the block to use 
```
unless enabled
    log dd trace not enabled
    call user code
end
```